### PR TITLE
HAAR-929: Starting to migrate to match template structure

### PR DESCRIPTION
**Pull request by brightonsbox**

### CODE REVIEW

File: backend/api/oauthApi.js [hunk 1]
```diff
@@ -22,12 +22,12 @@ const processPageResponse = (context) => (response) => {
...
  * @param client
  * @param {object} params
- * @param {string} params.clientId
- * @param {string} params.clientSecret
+ * @param {string} params.apiClientId
+ * @param {string} params.apiClientSecret
  * @param {string} params.url
  * @returns a configured oauthApi instance
  */
-const oauthApiFactory = (client, { clientId, clientSecret, url }) => {
```
Comment (brightonsbox) [inline backend/api/oauthApi.js:30]: slight rename to be consistent with naming in the template

File: backend/api/tokenVerificationApi.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
-const config = require('../config')
```
Comment (brightonsbox) [inline backend/api/tokenVerificationApi.js:1]: Mix of commonjs / es6 imports means I need to implement this workaround in js files for config

File: backend/api/tokenVerificationApi.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
+const config = require('../../server/config').default
 const log = require('../log')
 
 const tokenVerificationApiFactory = (client) => {
   const verifyToken = (context) => {
-    if (!config.apis.tokenverification.enabled) {
```
Comment (brightonsbox) [inline backend/api/tokenVerificationApi.js:6]: Some config renaming to match template

File: backend/data/redisClient.ts [hunk 4]
```diff
@@ -1,19 +1,19 @@
 import { createClient } from 'redis'
```
Comment (brightonsbox) [inline backend/data/redisClient.ts:1]: This now matches the template exactly

File: backend/middleware/setupWebSession.ts [hunk 5]
```diff
@@ -1,41 +1,46 @@
+import { v4 as uuidv4 } from 'uuid'
```
Comment (brightonsbox) [inline backend/middleware/setupWebSession.ts:1]: This now matches the template exactly

File: server.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+// Require app insights before anything else to allow for instrumentation of bunyan and express
```
Comment (brightonsbox) [inline server.ts:1]: This now matches the template exactly

File: server/app.ts [hunk 7]
```diff
@@ -0,0 +1,124 @@
+import express from 'express'
```
Comment (brightonsbox) [inline server/app.ts:1]: The name/location of this file matches template now, but the contents of this file need to be tackled next to bring in line

File: server/config.ts [hunk 8]
```diff
@@ -0,0 +1,100 @@
+import 'dotenv/config'
```
Comment (brightonsbox) [inline server/config.ts:1]: Now matches much more closely with the template

File: server/index.ts [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import promClient from 'prom-client'
```
Comment (brightonsbox) [inline server/index.ts:1]: Now mirrors the template, and introduces the metrics endpoint

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**